websockets/websockets_test.go

```diff
@@ -721,3 +725,50 @@ func TestOncloseDefineWithInvalidValue(t *testing.T) {
 	assert.Error(t, err)
 	assert.Contains(t, err.Error(), "a value for 'onclose' should be callable")
 }
+
+func TestCustomHeaders(t *testing.T) {
+	t.Parallel()
+	ts := newTestState(t)
+	sr := ts.tb.Replacer.Replace
+
+	mu := &sync.Mutex{}
+	collected := make(http.Header)
+
+	ts.tb.Mux.HandleFunc("/ws-echo-someheader", http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		responseHeaders := w.Header().Clone()
+		conn, err := (&websocket.Upgrader{}).Upgrade(w, req, responseHeaders)
+		if err != nil {
+			t.Fatalf("/ws-echo-someheader cannot upgrade request: %v", err)
+		}
+
+		mu.Lock()
+		collected = req.Header.Clone()
+		mu.Unlock()
+
+		err = conn.Close()
+		if err != nil {
+			t.Logf("error while closing connection in /ws-echo-someheader: %v", err)
+		}
+	}))
+
+	ts.state.CookieJar, _ = cookiejar.New(nil)
+	ts.ev.WaitOnRegistered()
```

Comment:
```suggestion
```

You might want to do this after, but even that seems not that useful :shrug: 

This should not be necessary. Did ytou run into problems?

Reply:
I just went and saw this was added to other tests as well and I am now wondering why it was added by you and @codebien :thinking:

Reply:
Ah, we don't need it here. Removing it!